lib/msf/core/post/linux/compile.rb

```diff
@@ -24,6 +36,8 @@ def live_compile?
     elsif datastore['COMPILER'] == 'clang' && has_clang?
       vprint_good 'clang is installed'
       return true
+    elsif datastore['COMPILER'] == 'Auto' && get_compiler != ''
+      return true
```

Comment:
Using the rails [`#present?`](https://apidock.com/rails/Object/present%3F) method will ensure that the value return from `#get_compiler` is a non empty string, it also accounts for nil.

```suggestion
    elsif datastore['COMPILER'] == 'Auto' && get_compiler.present?
      return true
```